tests/unit/mesh/test_boundary_factory.cxx

```diff
@@ -34,29 +34,22 @@ class TestBoundary : public BoundaryOp {
 class BoundaryFactoryTest : public ::testing::Test {
 public:
   BoundaryFactoryTest() {
-    if (mesh != nullptr) {
-      delete mesh;
-      mesh = nullptr;
-    }
+    delete mesh;
```

Comment:
Do we know it's always safe to delete the mesh here? Probably worth documenting why

Reply:
Yep -- if it's `nullptr`, it's always safe to delete. Otherwise, it's an actual value and if it's ours, we're good. If it's something we shouldn't delete, then we're in trouble as there's nothing we can do to check that.

Reply:
Ah I hadn't realised `nullptr` was safe to delete. Thanks!